tests/php/sync/test_class.jetpack-sync-callables.php

```diff
@@ -438,23 +439,33 @@ function test_subdomain_switching_to_www_does_not_cause_sync() {
 		$this->assertEquals( $original_site_url, $this->server_replica_storage->get_callable( 'site_url' ) );
 	}
 
-	function test_only_syncs_if_is_admin_and_not_cron() {
-		// non-admin
-		set_current_screen( 'front' );
-		$this->sender->do_sync();
-		$this->assertEquals( null, $this->server_replica_storage->get_callable( 'site_url' ) );
+	function test_sync_limited_set_of_callables_if_cron() {
+		$all_callables = array_keys( Defaults::get_callable_whitelist() );
+		$always_updated = Callables::ALWAYS_SEND_UPDATES_TO_THESE_OPTIONS;
 
-		set_current_screen( 'post-user' );
+		foreach ( $always_updated as $option ) {
+			if ( array_key_exists( $option, Callables::OPTION_NAMES_TO_CALLABLE_NAMES ) ) {
+				unset( $always_updated[ $option ] );
+				$always_updated[] = Callables::OPTION_NAMES_TO_CALLABLE_NAMES[ $option ];
```

Comment:
If we use `array_search` as suggested above, then we could directly update the existing element, rather than removing the previous one and inserting a new onne.

Reply:
Please see https://github.com/Automattic/jetpack/pull/13015/files#r304931686

Reply:
I might be misunderstanding how this works, but see https://github.com/Automattic/jetpack/pull/13015/files#r304934427